test/message/http_client_incoming_message_destroy.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
```

Comment:
I opted for testing that the process crashes, are there any alternatives?

Reply:
Process.on uncaught

Reply:
Thanks.

Reply:
Sorry for being annoying. While I was working on using `process.on()` in tests, I saw that the behavior in case of error is not the same as it is in version `15.4`. 

For example, this code:

```js
'use strict';

const common = require('../common');
const { createServer, get } = require('http');

const server = createServer(common.mustCall((req, res) => {
  res.writeHead(200);
  res.write('Part of res.');
}));

function onUncaught(error) {
  console.log('process', error)
}

process.on('uncaughtException', onUncaught)

server.listen(0, () => {
  get({
    port: server.address().port
  }, common.mustCall((res) => {
    res.socket.on('error', e => console.log('socket', e))
    res.destroy(new Error('Destroy test'));
  })).on('error', e => console.log('request', e));
});

```

on node `15.4`

gives this output:

```bash
request Error: Destroy test
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/parallel/test-http-client-incomingmessage-destroy.js:23:17)
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/common/index.js:377:15)
    at Object.onceWrapper (node:events:483:26)
    at ClientRequest.emit (node:events:376:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:660:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:126:17)
    at Socket.socketOnData (node:_http_client:526:22)
    at Socket.emit (node:events:376:20)
    at addChunk (node:internal/streams/readable:305:12)
    at readableAddChunk (node:internal/streams/readable:280:9)
socket Error: Destroy test
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/parallel/test-http-client-incomingmessage-destroy.js:23:17)
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/common/index.js:377:15)
    at Object.onceWrapper (node:events:483:26)
    at ClientRequest.emit (node:events:376:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:660:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:126:17)
    at Socket.socketOnData (node:_http_client:526:22)
    at Socket.emit (node:events:376:20)
    at addChunk (node:internal/streams/readable:305:12)
    at readableAddChunk (node:internal/streams/readable:280:9)
```

While on this branch:

```bash
request Error: Destroy test
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/parallel/test-http-client-incomingmessage-destroy.js:23:17)
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/common/index.js:377:15)
    at Object.onceWrapper (node:events:483:26)
    at ClientRequest.emit (node:events:376:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:648:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:126:17)
    at Socket.socketOnData (node:_http_client:514:22)
    at Socket.emit (node:events:376:20)
    at addChunk (node:internal/streams/readable:305:12)
    at readableAddChunk (node:internal/streams/readable:280:9)
socket Error: Destroy test
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/parallel/test-http-client-incomingmessage-destroy.js:23:17)
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/common/index.js:377:15)
    at Object.onceWrapper (node:events:483:26)
    at ClientRequest.emit (node:events:376:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:648:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:126:17)
    at Socket.socketOnData (node:_http_client:514:22)
    at Socket.emit (node:events:376:20)
    at addChunk (node:internal/streams/readable:305:12)
    at readableAddChunk (node:internal/streams/readable:280:9)
process Error: Destroy test
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/parallel/test-http-client-incomingmessage-destroy.js:23:17)
    at ClientRequest.<anonymous> (/home/dnlup/Dev/dnlup/node/test/common/index.js:377:15)
    at Object.onceWrapper (node:events:483:26)
    at ClientRequest.emit (node:events:376:20)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:648:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:126:17)
    at Socket.socketOnData (node:_http_client:514:22)
    at Socket.emit (node:events:376:20)
    at addChunk (node:internal/streams/readable:305:12)
    at readableAddChunk (node:internal/streams/readable:280:9)
```

It causes an `uncaughtException` even with a `error` handler attached to the `req` object. This is because now `res` is emitting an `error` and before it wasn't, I think. If we intend to keep this behavior, we should handle this in the `_http_client` (listen to `res.emit('error')` and emit an bubble up the event).

I  apologize if this statements are obvious to you, I am just trying to avoid big changes from the current implementation.